kasa/klaptransport.py

```diff
@@ -94,13 +95,18 @@ def __init__(
         *,
         credentials: Optional[Credentials] = None,
         timeout: Optional[int] = None,
+        port: Optional[int] = None,
     ) -> None:
         super().__init__(host=host)
 
-        self._credentials = credentials or Credentials(username="", password="")
+        self.credentials = credentials or Credentials(username="", password="")
+        self.timeout = timeout or self.DEFAULT_TIMEOUT
+        self.port = port or self.DEFAULT_PORT
+        self.http_client: httpx.AsyncClient = httpx.AsyncClient()
```

Comment:
I suppose storing all the connection parameters as I described above is up to debate, but here's how this `__init__` would work if we'd do that (maybe `params` is a too generic name here, but this is just to exemplify the idea):
```
def __init__(*, params: ConnectionParams):
    super().__init__(params=params)
    self._http_client: httpx.AsyncClient = params.httpx_client or httpx.AsyncClient()
```
Note, that the `BaseTransport` ctor can already assign the `host`, `port`, `credentials` variables like it is actually already doing.

Reply:
Done